pkg/client/client.go

```diff
@@ -86,25 +69,108 @@ func NewTableID(strID string) (TableID, error) {
 	return TableID(*tableID), nil
 }
 
+// Client is the Tableland client.
+type Client struct {
+	tblRPC      *rpc.Client
+	tblHTTP     *http.Client
+	tblContract *ethereum.Client
+	networkInfo NetworkInfo
+	relayWrites bool
+	wallet      *wallet.Wallet
+}
+
+type config struct {
+	networkInfo     *NetworkInfo
+	relayWrites     *bool
+	infuraAPIKey    string
+	alchemyAPIKey   string
+	local           bool
+	contractBackend bind.ContractBackend
+}
+
+// NewClientOption controls the behavior of NewClient.
+type NewClientOption func(*config)
+
+// NewClientNetworkInfo specifies network info.
+func NewClientNetworkInfo(networkInfo NetworkInfo) NewClientOption {
+	return func(ncc *config) {
+		ncc.networkInfo = &networkInfo
+	}
+}
+
+// NewClientRelayWrites specifies whether or not to relay write queries through the Tableland validator.
+func NewClientRelayWrites(relay bool) NewClientOption {
+	return func(ncc *config) {
+		ncc.relayWrites = &relay
+	}
+}
+
+// NewClientInfuraAPIKey specifies an Infura API to use when creating an EVM backend.
+func NewClientInfuraAPIKey(key string) NewClientOption {
+	return func(c *config) {
+		c.infuraAPIKey = key
+	}
+}
+
+// NewClientAlchemyAPIKey specifies an Alchemy API to use when creating an EVM backend.
+func NewClientAlchemyAPIKey(key string) NewClientOption {
+	return func(c *config) {
+		c.alchemyAPIKey = key
+	}
+}
+
+// NewClientLocal specifies that a local EVM backend should be used.
+func NewClientLocal() NewClientOption {
+	return func(c *config) {
+		c.local = true
+	}
+}
+
+// NewClientContractBackend specifies a custom EVM backend to use.
+func NewClientContractBackend(backend bind.ContractBackend) NewClientOption {
+	return func(c *config) {
+		c.contractBackend = backend
+	}
+}
+
 // NewClient creates a new Client.
-func NewClient(ctx context.Context, config Config) (*Client, error) {
+func NewClient(ctx context.Context, wallet *wallet.Wallet, opts ...NewClientOption) (*Client, error) {
```

Comment:
The client can now be created with options for which network, type of EVM backend to use (Alchemy, Infura, cusom) as well as a default setting for whether or not to relay writes.